consensus/polybft/stake_manager.go

```diff
@@ -334,39 +334,22 @@ func (s *stakeManager) UpdateValidatorSet(
 
 // getBlsKey returns bls key for validator from the supernet contract
 func (s *stakeManager) getBlsKey(address types.Address) (*bls.PublicKey, error) {
-	getValidatorFn := &contractsapi.GetValidatorCustomSupernetManagerFn{
-		Validator_: address,
-	}
-
-	encoded, err := getValidatorFn.EncodeAbi()
+	provider, err := s.blockchain.GetStateProviderForBlock(s.blockchain.CurrentHeader())
 	if err != nil {
 		return nil, err
 	}
 
-	response, err := s.rootChainRelayer.Call(
-		s.key.Address(),
-		ethgo.Address(s.supernetManagerContract),
-		encoded)
-	if err != nil {
-		return nil, fmt.Errorf("failed to invoke validators function on the supernet manager: %w", err)
-	}
-
-	byteResponse, err := hex.DecodeHex(response)
-	if err != nil {
-		return nil, fmt.Errorf("unable to decode hex response, %w", err)
-	}
+	stakeManagerContractContract := contract.NewContract(
+		ethgo.Address(s.stakeManagerContractAddr),
+		contractsapi.ValidatorSet.Abi, contract.WithProvider(provider),
+	)
 
-	decoded, err := validatorTypeABI.Decode(byteResponse)
+	rawResult, err := stakeManagerContractContract.Call("currentEpochId", ethgo.Latest)
```

Comment:
This is wrong. Why are you querying `currentEpochId`?

```suggestion
	rawResult, err := stakeManagerContractContract.Call("currentEpochId", ethgo.Latest)
```

Reply:
This is still wrong.